core/txpool/legacypool/legacypool.go

```diff
@@ -647,7 +647,7 @@ func (pool *LegacyPool) validateTx(tx *types.Transaction, local bool) error {
 		},
 		ExistingExpenditure: func(addr common.Address) *big.Int {
 			if list := pool.pending[addr]; list != nil {
-				return list.totalcost
+				return list.TotalCost(tx)
```

Comment:
This looks wrong, here we should check for the total cost of all pending transactions, not just the fee currency of `tx`. Or am I getting this worng?

Reply:
This is to check against the sender's balance for overdraft. So both the balance and the total cost has to be of the same currency as the tx being validated